File: cpp/webcam.cpp
#include <opencv2/opencv.hpp>

#include <iostream>

int main() {
    setLogLevel(cv::utils::logging::LOG_LEVEL_SILENT);

    cv::VideoCapture cap(0);

    if (!cap.isOpened()) {
        std::cerr << "Error: Unable to open camera" << std::endl;
        return 1;
    }

    namedWindow("Webcam", cv::WINDOW_AUTOSIZE);

    cv::Mat frame;
    while (true) {
        cap.read(frame);
        imshow("Webcam", frame);
        const auto key = cv::waitKey(1);
        if (key == 27 || key == 'q') {
            break;
        }
    }
}

File: python/webcam.py

import cv2

cap = cv2.VideoCapture(0)

if not cap.isOpened():
    print("Error: Unable to open camera")
    exit(1)

cv2.namedWindow("Webcam", cv2.WINDOW_AUTOSIZE)

while True:
    ret, frame = cap.read()

    frame = cv2.flip(frame, 1, frame) # flip
    cv2.imshow("Webcam", frame)

    if cv2.waitKey(1) in [ord('q'), 27]: # q or ESC
        break